data/pigui/libs/modal-win.lua

```diff
@@ -0,0 +1,81 @@
+-- Copyright © 2008-2020 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local ui = import 'pigui/pigui.lua'
+
+local modals = {}
+local modalStack = {}
+
+local ModalWindow = {}
+
+function ModalWindow.New(name, innerHandler, outerHandler, flags)
+	local self = {
+		name = name,
+		flags = flags or {"NoTitleBar", "NoResize", "AlwaysAutoResize", "NoMove"},
```

Comment:
I'd recommend you use the new WindowFlags helper class to create this set of flags once at file scope and simply set `flags = flags or defaultFlags`. This avoids the extra memory cost of the flags table, and prevents the table from having to be parsed once a frame!